app/controllers/act/event_categories_controller.rb

```diff
@@ -0,0 +1,183 @@
+  ##
+  #   Typ akce je škatulka zjednodušující práci s akcemi samotnými. Každá akce náleží právě jednomu typu akce.
+  #
+class Act::EventCategoriesController < ActController
+  ##
+  #   GET  /act/event_categories/index/
+  #
+  # Zobrazí všechny typy akcí
+  def index
+    if !(can? :read, Act::EventCategory)
+      render 'not_allowed', locals: { :desired => "read" }
+      return
+    end
+  end
+
+  ##
+  #  GET /act/event_categories/code
+  #
+  # Zobrazí daný typ akce
+  #
+  # *Params*
+  # code:: kód typu akce
+  #
+  def show
+    if !(can? :read, Act::EventCategory)
+      render 'not_allowed', locals: { :desired => "read" }
+      return
+    end
```

Comment:
použíj oneliner, zvlast kdyz je to v kazde metode. zvaz ze to vyhazis do ability.rb  a tam definujes co se muze a co ne.
```suggestion
      return render 'not_allowed', locals: { :desired => "read" } if ! can? :read, Act::EventCategory
   
```

Reply:
S onlinery je to dobrý point, opraveno (i v ostatních controllerech). V ability.rb to už v zásadě definované je (díky tomu můžu používat can? ... ). Používá to takhle, protože mi přijde hezčí vyhodit uživateli hlášku, co přesně nemůže, než nějakou generickou systémovou.

Reply:
jeste me napalo, ze to muzes jeste zkompaktnit kdyz misto `if` pouzijes `unless`, ale to necham na tobe.